crates/config/src/sections/passwords.rs

```diff
@@ -31,6 +31,23 @@ fn default_minimum_complexity() -> u8 {
     3
 }
 
+/// Configuration for REST password provider
+#[derive(Debug, Clone, Serialize, Deserialize, JsonSchema)]
+pub struct RestAuthProviderConfig {
+    /// The base URL where the identity service is implemented
+    pub url: String,
```

Comment:
How does that work in terms of authentication, etc? It would be useful to include links here to specifications for the API of such services, if there is any

Reply:
REST password providers are a kind of [Synapse modules](https://element-hq.github.io/synapse/latest/modules/password_auth_provider_callbacks.html). What they do is calling an identity provider on login (e. g. [MA1SD](https://github.com/ma1uta/ma1sd)), which itself implements a Matrix [Identity Service API](https://spec.matrix.org/v1.12/identity-service-api/).

It takes over all API login call (e. g. all authentications using the login API are authenticated against the identity provider), and creates accounts in Synapse database when an authentication is successful for a user which doesn't exist yet in the database.